### PR TITLE
Update Composer dependencies

### PR DESCRIPTION
**Pull request by 810**
Pull Request for Issue https://github.com/joomla/joomla-cms/pull/24442.

### Summary of Changes

composer updates

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -179,11 +179,15 @@ Desktop.ini
 /libraries/vendor/symfony/polyfill-php71/README.md
 /libraries/vendor/symfony/polyfill-php73/composer.json
 /libraries/vendor/symfony/polyfill-php73/README.md
+/libraries/vendor/symfony/polyfill-php73/Resources/stubs/JsonException.php
```
Comment (mbabker) [inline .gitignore]: This shouldn't be ignored.

**Comment by HLeithner** [comment]
Is it normal that we have the test classes in the autoloader?

e.x. Joomla\\Session\\Tests\\StorageCase

**Comment by mbabker** [comment]
It won't hurt anything.  The problem is after you do a `composer update` more often than not the gitignored files are going to be in your local filesystem, so when Composer generates the autoloaders because the files exist they'll be included in the maps.  Short of writing a script to execute at the `pre-autoload-dump` event to clean this up, you're not going to consistently fix this.

**Comment by mbabker** [comment]
Just for reference, it's a little better with the 4.0 build script because that explicitly runs `composer install --no-dev --no-autoloader` then cleans out non-shipped files before running `composer dump-autoload --no-dev --optimize`, but it still relies on regular review of the Composer installation and updating the script's `clean_checkout()` function.

**Comment by HLeithner** [comment]
thx for the info

**Comment by HLeithner** [comment]
@mbabker the phpmailer update prevent the use of urls in addAttachement, even this maybe would be a b/c break for us. I'm not sure show this function is exposed in joomla.

**Comment by HLeithner** [comment]
thx

**Comment by mbabker** [comment]
You're going to have to review CVE-2018-19296 then as that's the reason URLs aren't allowed in that method.

**Comment by HLeithner** [comment]
I already thought that this change is a security fix, thats the reason I merged it. Thx for the info.

**Pull request merged**